File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/PowerModifiedAmountMessage.ts
import {IMessage} from "Core/ObserverMessageKit";
import {AbstractPower} from "mods/ModTheSpire/Scripts/Power/AbstractPower";

export class PowerModifiedAmountMessage implements IMessage{
    public static readonly Id = "PowerModifiedAmountMessage";
    public constructor(public Power : AbstractPower,public Before:number,public After:number) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/PowerModifiedAmountMessage.cs
using ModTheSpire.Power;

namespace ModTheSpire.Event.Ids
{
    ///修改能量层数
    public class PowerModifiedAmountMessage
    {
        public AbstractPower Power;
        public int Before;
        public int After;

        public PowerModifiedAmountMessage(AbstractPower power, int before, int after)
        {
            Power = power;
            Before = before;
            After = after;
        }
    }
}